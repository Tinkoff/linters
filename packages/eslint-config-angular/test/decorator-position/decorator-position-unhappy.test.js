import ESlint from 'eslint';
import path from 'path';

describe('decorator-position / unhappy path', () => {
  const cli = new ESlint.CLIEngine({
    cwd: path.join(__dirname, '..'),
    useEslintrc: false,
    baseConfig: {
      extends: ['../decorator-position'],
    },
  });

  it('unhappy', () => {
    const codeframe = cli.getFormatter('codeframe');

    const report = cli.executeOnFiles([
      path.join(
        __dirname,
        './__fixtures__/decorator-position-unhappy.fixture.ts'
      ),
    ]);

    expect(codeframe(report.results)).toMatchSnapshot();
  });
});
