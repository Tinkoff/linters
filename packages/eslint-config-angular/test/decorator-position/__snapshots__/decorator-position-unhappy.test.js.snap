// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`decorator-position / unhappy path unhappy 1`] = `
"error: Expected @Input to be on the line above (decorator-position/decorator-position) at packages/eslint-config-angular/test/decorator-position/__fixtures__/decorator-position-unhappy.fixture.ts:10:3:
   8 | 
   9 | class B {
> 10 |   @Input() name: any;
     |   ^
  11 | 
  12 |   @pure() foo() {}
  13 | }


error: Expected @pure to be on the line above (decorator-position/decorator-position) at packages/eslint-config-angular/test/decorator-position/__fixtures__/decorator-position-unhappy.fixture.ts:12:3:
  10 |   @Input() name: any;
  11 | 
> 12 |   @pure() foo() {}
     |   ^
  13 | }
  14 | 


2 errors found.
2 errors potentially fixable with the \`--fix\` option."
`;
